lib/srv/db/snowflake/models.go

```diff
@@ -0,0 +1,145 @@
+/*
+
+ Copyright 2022 Gravitational, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+
+*/
+
+package snowflake
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+// loginRequest is the payload sent to /queries/v1/query-request endpoint.
+type loginRequest struct {
+	Data struct {
+		ClientAppID             string          `json:"CLIENT_APP_ID"`
+		ClientAppVersion        string          `json:"CLIENT_APP_VERSION"`
+		SvnRevision             string          `json:"SVN_REVISION"`
+		AccountName             string          `json:"ACCOUNT_NAME"`
+		LoginName               string          `json:"LOGIN_NAME,omitempty"`
+		Password                string          `json:"PASSWORD,omitempty"`
+		RawSAMLResponse         string          `json:"RAW_SAML_RESPONSE,omitempty"`
+		ExtAuthnDuoMethod       string          `json:"EXT_AUTHN_DUO_METHOD,omitempty"`
+		Passcode                string          `json:"PASSCODE,omitempty"`
+		Authenticator           string          `json:"AUTHENTICATOR,omitempty"`
+		SessionParameters       json.RawMessage `json:"SESSION_PARAMETERS,omitempty"`
+		ClientEnvironment       json.RawMessage `json:"CLIENT_ENVIRONMENT"`
+		BrowserModeRedirectPort string          `json:"BROWSER_MODE_REDIRECT_PORT,omitempty"`
+		ProofKey                string          `json:"PROOF_KEY,omitempty"`
+		Token                   string          `json:"TOKEN,omitempty"`
+	} `json:"data"`
+}
+
+// loginResponse is the payload returned by the /queries/v1/query-request endpoint.
+type loginResponse struct {
+	// use map here to not remove any fields when marshaling back.
+	Data map[string]interface{} `json:"data"`
```

Comment:
use json.RawMessage  ? 

then in `getTokens` you can define your own type with fileds liek  `validityInSeconds` `masterToken` and Unmarshal it to the custom type instead of calling `getField("validityInSeconds")`